tests/Test/AbstractTransformerTestCase.php

```diff
@@ -27,26 +27,41 @@ protected function doTest($source, array $expectedTokens = array(), array $obser
     {
         $tokens = Tokens::fromCode($source);
 
-        $this->assertSame(
-            count($expectedTokens),
-            array_sum(array_map(
-                function ($item) {
-                    return count($item);
-                },
-                $tokens->findGivenKind(array_unique(array_merge($observedKinds, array_values($expectedTokens))))
-            )),
-            'Number of expected tokens does not match actual token count.'
-        );
-
-        foreach ($expectedTokens as $index => $tokenId) {
+        if (count($observedKinds)) {
```

Comment:
I won't agree with this condition @julienfalque 
you just relaxed the `doTest` method to sometimes make less assertion.

to extra simpily the change done here, assuming `$observedKinds` are empty in given execution, this change is:
```diff
+ if (count($observedKinds)) {
     $this->assertCount(0, $observedKinds)
+}
```